generator/README.md

```diff
@@ -0,0 +1,824 @@
+ * overlay currently has a higher priority than this alert. An app must not send an alert without waiting at least
+ * the amount of time dictated.
+ * {"default_value": null, "max_value": 2000000000, "min_value": 0}
+ *
+ * @since SDL 2.0.0
+ *
+ * Optional, UInt32
+ */
+@property (nullable, strong, nonatomic) NSNumber<SDLUInt> *tryAgainTime;
+
+@end
+
+NS_ASSUME_NONNULL_END
+```
+
+```objc
+//  SDLAlertResponse.m
+//
+
+
+#import "SDLAlertResponse.h"
+#import "NSMutableDictionary+Store.h"
+#import "SDLRPCFunctionNames.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation SDLAlertResponse
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+- (instancetype)init {
+    self = [super initWithName:SDLRPCFunctionNameShowAppMenu];
+    if (!self) { return nil; }
+
+    return self;
+}
+#pragma clang diagnostic pop
+
+- (void)setTryAgainTime:(nullable NSNumber<SDLUInt> *)tryAgainTime {
+    [self.parameters sdl_setObject:tryAgainTime forName:SDLRPCParameterNameTryAgainTime];
+}
+
+- (nullable NSNumber<SDLUInt> *)tryAgainTime {
+    return [self.parameters sdl_objectForName:SDLRPCParameterNameTryAgainTime ofClass:NSNumber.class error:nil];
+}
+
+@end
+
+NS_ASSUME_NONNULL_END
+```
+
+##### Notification Functions (SDLRPCNotification)
+
+Another example is the class `SDLOnAppInterfaceUnregistered`. It is declared in the XML as follows:
+
+```xml
+<function name="OnAppInterfaceUnregistered" functionID="OnAppInterfaceUnregisteredID" messagetype="notification" since="1.0">
+	<param name="reason" type="AppInterfaceUnregisteredReason" mandatory="true">
+	    <description>See AppInterfaceUnregisteredReason</description>
+	</param>
+</function>
+```
+
+The corresponding Objective-C class `SDLOnAppInterfaceUnregistered` looks as following.
+
+```objc
+//  SDLOnAppInterfaceUnregistered.h
+
+#import "SDLRPCNotification.h"
+
+@class SDLAppInterfaceUnregisteredReason;
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * @since SDL 1.0.0
+ */
+@interface SDLOnAppInterfaceUnregistered : SDLRPCNotification
+
+/**
+ * @param reason - @(reason)
+ * @return A SDLOnAppInterfaceUnregistered object
+ */
+- (instancetype)initWithReason:(SDLAppInterfaceUnregisteredReason)reason;
+
+/**
+ * See AppInterfaceUnregisteredReason
+ *
+ * Required, SDLAppInterfaceUnregisteredReason 
+ */
+@property (strong, nonatomic) SDLAppInterfaceUnregisteredReason reason;
+
+@end
+
+NS_ASSUME_NONNULL_END
+```
+
+*Note: though the `SDLAppInterfaceUnregisteredReason reason` declared as a property but implemented as custom getter and setter*
+
+```objc
+//  SDLOnAppInterfaceUnregistered.m
+
+#import "SDLOnAppInterfaceUnregistered.h"
+#import "NSMutableDictionary+Store.h"
+#import "SDLAppInterfaceUnregisteredReason.h"
+#import "SDLRPCParameterNames.h"
+#import "SDLRPCFunctionNames.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation SDLOnAppInterfaceUnregistered
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
```

Comment:
Notification inits are currently deprecated, but they should not be. Can we generate them as not deprecated please?

Reply:
Do you talking about this lines:
```objc
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wdeprecated-declarations"
... 
#pragma clang diagnostic pop
```
@joeljfischer the same lines are present in all already existing `functions`, do you want to delete this lines from template?

Reply:
I apologize, that was a mistake on my part. I looked back into it again to confirm, and you're correct, they should not be deleted.